Core/include/Acts/EventData/TrackStateType.hpp

```diff
@@ -41,7 +41,20 @@ class TrackStateType {
       std::numeric_limits<std::make_unsigned<raw_type>::type>::digits;
   /// Constructor from a reference to the underlying value container
   /// @param raw the value container
-  TrackStateType(raw_type& raw) : m_raw{&raw} {}
+  TrackStateType(raw_type& raw) : m_raw{&raw} { assert(m_raw != nullptr); }
```

Comment:
Isn't this redundant? The reference can't be to nullptr.

Reply:
Hm can't you just do `TrackStateType(*nullptr)` and end up with `m_raw = nullptr`? Because in principle the `nullptr` will not be dereferenced but just captured by reference.

Reply:
or even use the default constructor

Reply:
I think the default constructor is implicitly deleted because there is one with an argument but one could also explicitly delete it which is best practice in C++ I believe https://en.cppreference.com/w/cpp/language/rule_of_three